packages/wrangler/src/config.ts

```diff
@@ -485,3 +487,160 @@ export async function validateConfig(
 
   return results;
 }
+
+/**
+ * Process the environments (`env`) specified in the `config`.
+ *
+ * The environments configuration is complicated since each environment is a customized version of the main config.
+ * Some of the configuration can be inherited from the main config, while other configuration must replace what is in the main config.
+ *
+ * This function ensures that each environment is set up correctly with inherited configuration, as necessary.
+ * It will log a warning if an environment is missing required configuration.
+ */
+export function processEnvironmentConfig(config: Config) {
```

Comment:
"Process" sounds a bit generic. This function - normalises some environment fields, and - validates some others. So how about the verbose `normaliseAndValidateEnvironmentsConfig()`?

Reply:
Sure. You can tell I was having difficulty naming this one 😃